File: CocomeRegister/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CocomeStore.Models
{
    /// <summary>
    /// class <c>Product</c>
    /// </summary>
    public class Product
    {
        [ScaffoldColumn(false)]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }
        
        [Required]
        public float Price { get; set; }

        [Required]
        public float SalePrice { get; set; }

        public string Description { get; set; }

        public string ImageUrl { get; set; }

        [Required]
        public int ProviderId { get; set; }

        public Provider Provider { get; set; }
    }
}

File: CocomeRegister/ClientApp/src/services/Product.ts
import { Store } from "./Store";

export interface Product {
    id: number;
    name: string;
    price: number;
    salePrice: number;
    description: string | undefined;
    imageUrl: string | undefined;
}

export interface StockItem {
    id: number;
    product: Product;
    stock: number;
    store: Store;
    minimum: number | undefined;
}
